src/AlevinUtils.cpp

```diff
@@ -570,6 +627,51 @@ namespace alevin {
         std::not2(std::equal_to<std::string::value_type>()));
     }
 
+    void addPadding(std::string& seq, uint32_t max, std::string padBases, unsigned int padLen) {
+      int diff = max - seq.length() + 1; // add one base if the length is same to avoid erroneous collisions
+      int rep = diff / padLen;
```

Comment:
I imagine we want to avoid a `/` and `%` in tight loops.  Is there another way we can do this padding?

Reply:
I have updated the function. I'm wondering what the reason for avoiding `/` and `%` is?

Reply:
/ and % are much much slower than * and +. So if possible, one should avoid them in critical paths (unless the divisor or the modulo is known at compile time, in which case the compiler computes the (integer) inverse of the divisor and replaces the division by a multiplication).